datafusion/sql/src/statement.rs

```diff
@@ -128,69 +128,91 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
-                                })
-                                .collect::<Vec<_>>();
-                            LogicalPlanBuilder::from(plan.clone())
-                                .project(project_exprs)?
-                                .build()?
-                        } else {
-                            plan
-                        };
-
-                        Ok(LogicalPlan::CreateMemoryTable(CreateMemoryTable {
-                            name: self.object_name_to_table_reference(name)?,
-                            input: Arc::new(plan),
-                            if_not_exists,
-                            or_replace,
-                        }))
-                    }
+                            })
+                            .collect::<Vec<_>>();
+                        LogicalPlanBuilder::from(plan.clone())
+                            .project(project_exprs)?
+                            .build()?
+                    } else {
+                        plan
+                    };
+
+                    Ok(LogicalPlan::CreateMemoryTable(CreateMemoryTable {
+                        name: self.object_name_to_table_reference(name)?,
+                        primary_key: vec![],
+                        input: Arc::new(plan),
+                        if_not_exists,
+                        or_replace,
+                    }))
+                }
 
-                    None => {
-                        let schema = self.build_schema(columns)?.to_dfschema_ref()?;
-                        let plan = EmptyRelation {
-                            produce_one_row: false,
-                            schema,
-                        };
-                        let plan = LogicalPlan::EmptyRelation(plan);
-
-                        Ok(LogicalPlan::CreateMemoryTable(CreateMemoryTable {
-                            name: self.object_name_to_table_reference(name)?,
-                            input: Arc::new(plan),
-                            if_not_exists,
-                            or_replace,
-                        }))
-                    }
+                None => {
+                    let pk: Option<Vec<Ident>> = constraints
+                        .iter()
+                        .filter_map(|c: &TableConstraint| match c {
+                            TableConstraint::Unique {
+                                columns,
+                                is_primary,
+                                ..
+                            } => match is_primary {
+                                true => Some(columns),
+                                false => None,
+                            },
+                            TableConstraint::ForeignKey { .. } => None,
+                            TableConstraint::Check { .. } => None,
+                            TableConstraint::Index { .. } => None,
+                            TableConstraint::FulltextOrSpatial { .. } => None,
```

Comment:
Should I throw the original error if these are set?

Reply:
Yes I think throwing an error is better than silently ignoring them